src/Domain/FunctionsOfTime/SettleToConstantQuaternion.cpp

```diff
@@ -0,0 +1,136 @@
+
+#include "Domain/FunctionsOfTime/SettleToConstantQuaternion.hpp"
+
+#include <cmath>
+#include <limits>
+#include <memory>
+
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace {
+template <size_t MaxDerivReturned>
+std::array<double, MaxDerivReturned + 1> quat_norm_and_derivs(
+    const std::array<DataVector, MaxDerivReturned + 1>&
+        unnormalized_func_and_derivs) {
+  std::array<double, MaxDerivReturned + 1> result =
+      make_array<MaxDerivReturned + 1>(0.0);
+  gsl::at(result, 0) =
+      sqrt(sum(square(gsl::at(unnormalized_func_and_derivs, 0))));
+  if constexpr (MaxDerivReturned > 0) {
+    gsl::at(result, 1) = sum(gsl::at(unnormalized_func_and_derivs, 0) *
+                             gsl::at(unnormalized_func_and_derivs, 1)) /
+                         gsl::at(result, 0);
+    if constexpr (MaxDerivReturned > 1) {
+      gsl::at(result, 2) =
+          sum(square(gsl::at(unnormalized_func_and_derivs, 1)) +
+              gsl::at(unnormalized_func_and_derivs, 0) *
+                  gsl::at(unnormalized_func_and_derivs, 2)) /
+          gsl::at(result, 0);
+    }
+  }
+  return result;
+}
+};  // namespace
+
+namespace domain::FunctionsOfTime {
+SettleToConstantQuaternion::SettleToConstantQuaternion(
+    const std::array<DataVector, 3>& initial_func_and_derivs,
+    const double match_time, const double decay_time)
+    : unnormalized_function_of_time_(initial_func_and_derivs, match_time,
+                                     decay_time),
+      match_time_(match_time),
+      decay_time_(decay_time),
+      size_(initial_func_and_derivs[0].size()) {
+  if (size_ != 4) {
+    ERROR(
+        "SettleToConstantQuaternion requires the initial function and its time "
+        "derivatives to be quaternions stored as DataVectors of size 4, not "
+        "size "
+        << size_);
+  }
+}
+
+std::unique_ptr<FunctionOfTime> SettleToConstantQuaternion::get_clone() const {
+  return std::make_unique<SettleToConstantQuaternion>(*this);
+}
+
+template <size_t MaxDerivReturned>
+std::array<DataVector, MaxDerivReturned + 1>
+SettleToConstantQuaternion::func_and_derivs(const double t) const {
+  static_assert(MaxDerivReturned < 3, "The maximum available derivative is 2.");
+
+  std::array<DataVector, MaxDerivReturned + 1> result =
+      make_array<MaxDerivReturned + 1>(DataVector(size_, 0.0));
+  if constexpr (MaxDerivReturned == 0) {
+    result = unnormalized_function_of_time_.func(t);
+  } else if constexpr (MaxDerivReturned == 1) {
+    result = unnormalized_function_of_time_.func_and_deriv(t);
+  } else if constexpr (MaxDerivReturned == 2) {
+    result = unnormalized_function_of_time_.func_and_2_derivs(t);
+  }
+  std::array<double, MaxDerivReturned + 1> norm_and_derivs =
+      quat_norm_and_derivs<MaxDerivReturned>(result);
+  gsl::at(result, 0) /= gsl::at(norm_and_derivs, 0);
```

Comment:
[optional] You can avoid repeating the divide by including `#include "Utilities/StdArrayHelpers.hpp` doing
```cpp
result /= gsl::at(norm_and_derivs, 0);
```
This will divide everything at once.